framework/Databases.py

```diff
@@ -222,402 +243,87 @@ def getEndingGroupNames(self):
-    else:
-      if allOutParam:
-        for key in histVar[1]['outputSpaceHeaders']:
-          outDict[key] = np.atleast_1d(np.array(interp1d(histVar[0][:,pivotIndex], histVar[0][:,histVar[1]['outputSpaceHeaders'].index(key)], kind='linear')(outputPivotVal)))
-      else:
-        for key in outParam:
-          if key in histVar[1]['outputSpaceHeaders']:
-            outDict[key] = np.atleast_1d(np.array(interp1d(histVar[0][:,pivotIndex], histVar[0][:,histVar[1]['outputSpaceHeaders'].index(key)], kind='linear')(outputPivotVal)))
-          else:
-            self.raiseAnError(IOError,"the parameter " + key + " has not been found")
-    # Return tuple of dictionaries containing the HistorySet
-    return (copy.copy(inDict),copy.copy(outDict),copy.copy(metaDict))
-
-  def retrieveData(self,attributes):
+      Method to obtain a realization from the data, either by index (e.g. realization number) or matching value.
+      Either "index" or "matchDict" must be supplied. (NOTE: now just "index" can be supplied)
+      @ In, index, int or str, optional, number of row to retrieve (by index, not be "sample") or group name
+      @ In, matchDict, dict, optional, {key:val} to search for matches
+      @ In, tol, float, optional, tolerance to which match should be made
+      @ Out, index, int, optional, index where found (or len(self) if not found), only returned if matchDict
+      @ Out, rlz, dict, realization requested (None if not found)
     """
-      Function interface for retrieving a Point or PointSet or History from the HDF5 database
-      @ In, attributes, dict, options (metadata muste be appended to the root group)
-      @ Out, data, tuple, tuple in which the first position is a dictionary of numpy arrays (input variable)
-      and the second is a dictionary of the numpy arrays (output variables).
-    """
-    if attributes['type'] == 'PointSet':
-      data = self.__retrieveDataPointSet(attributes)
-    elif attributes['type'] == 'HistorySet':
-      listhistIn  = {}
-      listhistOut = {}
-      listhistMeta= {}
-      endGroupNames = self.getEndingGroupNames()
-      for index in range(len(endGroupNames)):
-        attributes['history'] = endGroupNames[index]
-        tupleVar = self.__retrieveDataHistory(attributes)
-        # dictionary of dictionary key = i => ith history ParameterValues dictionary
-        listhistIn[index]  = tupleVar[0]
-        listhistOut[index] = tupleVar[1]
-        listhistMeta[index]= tupleVar[2]
-        del tupleVar
-      data = (listhistIn,listhistOut,listhistMeta)
+    # matchDict not implemented for Databases
+    assert (matchDict is None)
+    if (not self.exist) and (not self.built):
+      self.raiseAnError(Exception,'Can not retrieve a realization from Database' + self.name + '.It has not been built yet!')
+    if type(index).__name__ == 'int': allRealizations = self.database.retrieveAllHistoryNames()
+    if type(index).__name__ == 'int' and index > len(allRealizations):
```

Comment:
Similarly, no need to fix, but could these be `type(index) == int` instead of checking the string name?